modules/terraform/options.go

```diff
@@ -66,29 +66,15 @@ type Options struct {
 }
 
 // Clone makes a deep copy of most fields on the Options object and returns it.
+//
+// NOTE: options.SshAgent and options.Logger can NOT be deep copied (e.g., the SshAgent struct contains channels and
+// listeners that can't be meaningfully copied), so the original values are retained.
```

Comment:
NIT: worth combining the newly added comment with the previous to say that Clone makes a deep copy of all fields except SshAgent and Logger?

Reply:
I think it works either way, and have a slight preference for the current version, as it makes the intention of the method clear in one sentence, and uses a separate sentence to try to explain the limitations... So I'll leave as-is.